lib/charms/layer/apache_bigtop_base.py

```diff
@@ -491,6 +491,31 @@ def get_hadoop_version():
     return parts[1]
 
 
+def get_package_version(pkg=None):
+    """
+    Return a version string for a given package name.
+
+    :param: str pkg: package name as known by the package manager
+    :returns: version string or None
+    """
+    ver_str = None
+    if pkg:
+        distro = lsb_release()['DISTRIB_ID'].lower()
+        if distro == 'ubuntu':
+            # NB: we cannot use the charmhelpers.fetch.apt_cache nor the
+            # apt module from the python3-apt deb as they are only available
+            # as system packages. Any charm with use_system_packages=False in
+            # layer.yaml would fail. Use dpkg-query instead.
+            cmd = ['dpkg-query', '--show', r'--showformat=${Version}', pkg]
+            try:
+                ver_str = subprocess.check_output(cmd).strip().decode()
+            except subprocess.CalledProcessError as e:
+                hookenv.log(
+                    'Error getting package version: {}'.format(e.output),
+                    hookenv.ERROR)
+    return ver_str
```

Comment:
I opted for returning None in case something prevented dpkg from running (eg, something else has the dpkg lock).  Since the main use is just for status output, I didn't want to put the charm in a failed state just because we weren't able to determine the pkg version.

I figured the caller would check for a valid return, and if it was None, they could decide whether to set something (like "unkown"), retry later, or do nothing.